src/main/java/org/wso2/carbon/apimgt/securityenforcer/utils/SecurityUtils.java

```diff
@@ -105,10 +100,22 @@ public static JSONArray getTransportHeaders(org.apache.axis2.context.MessageCont
                 }
             }
 
-            for (String headerKey : headerKeysSet) {
-                String headerValue = transportHeadersMap.get(headerKey);
-                transportHeadersArray.add(addObj(headerKey, headerValue));
+            if (ServiceReferenceHolder.getInstance().getSecurityHandlerConfig().getLimitTransportHeaders().isEnable()) {
+                Set<String> allowedTransportHeadersKeySet = ServiceReferenceHolder.getInstance()
```

Comment:
Shall we take ServiceReferenceHolder.getInstance().getSecurityHandlerConfig().getLimitTransportHeaders() outside the if block and assign to a variable and resuse?

Reply:
Fixed with https://github.com/wso2-extensions/apim-handler-pingai/pull/22